### DISCUSSION TITLE
Is there a `yarn why` equivalence?

### DISCUSSION THREAD

**Discussion by unional** [pnpm/pnpm#5571] [Q&A] [answered]
This is one this sorely missed in `pnpm`.

There are many times I want to figure out why there are some extra packages get added to `node_modules/.pnpm/cache`.

Especially `@types/*` because having multiple version of the same `@types` package means the typings is messed up and you can't compile.

Using `@types/react` as an example:

`pnpm list -r @types/react` only gives you whats declared in the package.json, not all the versions (and hash) get added to `node_modules/.pnpm/cache`.

**Comment by zkochan** [comment, marked as answer]
There is [pnpm why](https://pnpm.io/cli/why)

**Reply by unional** [reply]
Yes, you are correct. Originally I thought it was broken as it does not do anything, but turns out I need to do `pnpm why -r <pkg>` for monorepo.
